code/controllers/subsystems/records.dm

```diff
@@ -239,56 +237,46 @@
 			var/depDat
 			for(var/list/item in depI)
 				depDat += "<li><strong>[item["name"]]</strong> - [item["rank"]] ([item["active"]])</li>"
-			dat += "<h3>[nameMap[dep]]</h3><ul>[depDat]</ul>"
+			dat += "<h3>[dep]</h3><ul>[depDat]</ul>"
 	return dat
 
 /datum/controller/subsystem/records/proc/get_manifest_list()
 	if(manifest.len)
 		return manifest
-	manifest = list(
-		"heads" = list(),
-		"sec" = list(),
-		"eng" = list(),
-		"med" = list(),
-		"sci" = list(),
-		"car" = list(),
-		"civ" = list(),
-		"bot" = list(),
-		"misc" = list()
-	)
-	var/positions = list(
-		"heads" = command_positions,
-		"sec" = security_positions,
-		"eng" = engineering_positions,
-		"med" = medical_positions,
-		"sci" = science_positions,
-		"car" = cargo_positions,
-		"civ" = civilian_positions,
-		"bot" = nonhuman_positions
-	)
+	manifest = list()
+	for(var/department in positions_by_department)  // prepare empty manifest
+		manifest[department] = list()
 	for(var/datum/record/general/t in records)
 		var/name = sanitize(t.name, encode = FALSE)
 		var/rank = sanitize(t.rank, encode = FALSE)
 		var/real_rank = make_list_rank(t.real_rank)
 
 		var/isactive = t.physical_status
-		var/department = 0
-		var/depthead = 0            // Department Heads will be placed at the top of their lists.
-
-		for(var/positionType in positions)
-			var/typesPositions = positions[positionType]
-			if(real_rank in typesPositions)
-				manifest[positionType][++manifest[positionType].len] = list("name" = name, "rank" = rank, "active" = isactive)
-				department = 1
-				if ((depthead || rank == "Captain") && manifest[positionType].len != 1)
-					manifest[positionType].Swap(1, manifest[positionType].len)
-					manifest[positionType][1]["head"] = TRUE
-				if(positionType == "head")
-					depthead = 1
-
-		if(!department && !(name in manifest["heads"]))
-			manifest["misc"][++manifest["misc"].len] = list("name" = name, "rank" = rank, "active" = isactive)
+		var/dept = null
+		var/depthead = FALSE
+
+		for(var/department in positions_by_department)
+			if(!(real_rank in positions_by_department[department])) // search for a department with that rank
+				continue
+			dept = department
+			if(department == DEPARTMENT_COMMAND)
+				depthead = TRUE
+				continue
+			// because Command is first we have guarantee that the depthead var has been assigned and we have found the actual dept so we can safely break
```

Comment:
I don't really like that this all relies on the ordering of `positions_by_department`. 

We have this 'department_flag` thing, too, that we're ignoring. 

Could we define heads as being e.g. `flag = CIVILIAN | HEAD` and check that var instead?

I don't know why we have both 'department' and 'department_flag'

Reply:
> I don't really like that this all relies on the ordering of positions_by_department.

Me neither, but as stated above Command is just ... special, and I'm not sure it would be easy to deal with that.

---

I'll take a look at the flags. But I think we need the constants regardless, for output procs (like the manifest) if nothing else.

Reply:
Fixed by asking SSJobs for the job instead of this nightmare.